src/test.js

```diff
@@ -30,13 +30,24 @@ export default function Test( settings ) {
 	++Test.count;
 
 	this.expected = null;
-	extend( this, settings );
 	this.assertions = [];
 	this.semaphore = 0;
 	this.module = config.currentModule;
 	this.stack = sourceFromStacktrace( 3 );
 	this.steps = [];
 
+	// If a module is skipped, all its tests and the tests of the child suites
+	// should be treated as skipped even if they are defined as `only` or `todo`.
```

Comment:
I tested out the hypothetical scenario with Mocha and I like the way they handle it. Basically, if you have a `skip` module and an `only` test, then the single test is run but it is also skipped, which makes sense. I think we should follow that pattern.

But, I agree that `skip` should take precedence over `todo`.